iabtcf-decoder/src/main/java/com/iabtcf/v2/PublisherRestriction.java

```diff
@@ -32,11 +33,15 @@
     private final RestrictionType restrictionType;
     private final IntIterable vendorIds;
 
-    public PublisherRestriction(
+    private PublisherRestriction(
             int purposeId, RestrictionType restrictionType, IntIterable vendorIds) {
         Objects.requireNonNull(vendorIds);
         Objects.requireNonNull(restrictionType);
 
+        if (purposeId <= 0) {
```

Comment:
can you reference the spec for this? also, can you move this check down to the builder?

Reply:
I am not able to find any spec for this check. The spec just defines purposes. I just copied this check from the existing PublisherRestrictionEntry class

Reply:
Moved it to the build method.